src/python/pants/backend/shell/target_types.py

```diff
@@ -442,6 +466,41 @@ class ShellCommandTarget(Target):
     )
 
 
+class ShellRunInSandboxTarget(Target):
+    alias = "experimental_run_in_sandbox"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        RunInSandboxRunnableField,
+        RunInSandboxArgumentsField,
+        ShellCommandExecutionDependenciesField,
+        ShellCommandOutputDependenciesField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
```

Comment:
Should this be using the new style from #17744?

```suggestion
        ShellCommandOutputFilesField,
        ShellCommandOutputDirectoriesField,
```